xdsl/dialects/gpu.py

```diff
@@ -244,14 +245,3 @@ class ThreadIdOp(Operation):
     def get(dim: DimensionAttr) -> ThreadIdOp:
         return ThreadIdOp.build(result_types=[IndexType()],
                                 attributes={"dimension": dim})
-
-
-#_GPUAttr has to be registered instead of DimensionAttr and AllReduceOperationAttr here.
-# This is a hack to fit MLIR's syntax in xDSL's way of parsing attributes, without making GPU builtin.
-# Hopefully MLIR will parse it in a more xDSL-friendly way soon, so all that can be factored in proper xDSL
-# atrributes.
-GPU = Dialect([
-    BarrierOp, BlockDimOp, BlockIdOp, GlobalIdOp, GridDimOp, LaneIdOp,
-    ModuleOp, ModuleEndOp, NumSubgroupsOp, SetDefaultDeviceOp, SubgroupIdOp,
-    SubgroupSizeOp, ThreadIdOp
-], [_GPUAttr])
```

Comment:
Could you please check that the dialect parsed here is the same as the one manually constructed? The point was that `_GPUAttr` is special